tests/unit/dataactvalidator/test_sf133_content.py

```diff
@@ -0,0 +1,161 @@
+                row['availability_type_code'].strip() if row['availability_type_code'].strip() else ' ',
+                row['main_account_code'] if row['main_account_code'] else '0000',
+                row['sub_account_code'] if row['sub_account_code'] else '000'
+            ]), axis=1)
+            data['amount'] = data['amount'].astype(float)
+
+            data = load_sf133.fill_blank_sf133_lines(data)
+
+            # remove all lines not included in validation
+            sf_133_validation_lines = [
+                '1000', '1010', '1011', '1012', '1013', '1014', '1015', '1016', '1017', '1018', 
+                '1019', '1020', '1021', '1022', '1023', '1024', '1025', '1026', '1027', '1028', 
+                '1029', '1030', '1031', '1032', '1033', '1034', '1035', '1036', '1037', '1038', 
+                '1039', '1040', '1041', '1042', '1160', '1180', '1260', '1280', '1340', '1440', 
+                '1540', '1640', '1750', '1850', '1910', '2190', '2490', '2500'
+            ]
+            data = data[(data.line.isin(sf_133_validation_lines) & (data.amount!=0))]
+
+            # create list of unique TAS
+            data.sort_values(by=['tas'], inplace=True)
+            data.set_index(keys=['tas'], drop=False, inplace=True)
+            all_tas=data['tas'].unique().tolist()
+
+            # iterate through TAS list
+            for tas in all_tas:
+                current_tas = data.loc[data.tas==tas]
+
+                sum_one = 0
+                validate_one = 0
+                validate_one_lines = [
+                    '1000', '1010', '1011', '1012', '1013', '1014', '1015', '1016', '1017', 
+                    '1018', '1019', '1020', '1021', '1022', '1023', '1024', '1025', '1026', 
+                    '1027', '1028', '1029', '1030', '1031', '1032', '1033', '1034', '1035', 
+                    '1036', '1037', '1038', '1039', '1040', '1041', '1042', '1160', '1180', 
+                    '1260', '1280', '1340', '1440', '1540', '1640', '1750', '1850'
+                ]
+
+                sum_two = 0
+                validate_two = 0
+                validate_two_lines = ['2190', '2490']
+                for index, row in current_tas.iterrows():
+                    if row.line in validate_one_lines:
+                        sum_one += row.amount
+                    elif row.line=='1910':
+                        validate_one = float("{0:.2f}".format(row.amount))
+                    elif row.line in validate_two_lines:
+                        sum_two += row.amount
+                    elif row.line=='2500':
+                        validate_two = float("{0:.2f}".format(row.amount))
+                sum_one = float("{0:.2f}".format(sum_one))
+                sum_two = float("{0:.2f}".format(sum_two))
+
+                validation_one = sum_one==validate_one
+                validation_two = sum_two==validate_two
+                validation_three = validate_one==validate_two
+
+                tas_m = TAS_RE.match(tas)
```

Comment:
It feels off to smash together a bunch of fields to create the `tas`, then parse them out with a regular expression. Would it make sense to use a representative row or something like it to access the original field names?